lib/jar_dependencies.rb

```diff
@@ -275,11 +270,16 @@ def require_jar(group_id, artifact_id, *classifier_version)
     end
 
     def warn(msg = nil)
-      Kernel.warn(msg || yield) unless quiet? && !verbose?
+      return if (verbose? == nil && quiet?) || (verbose? == false && !debug?)
```

Comment:
```ruby
return if quiet? || !(verbose? || debug?)
```

maybe?

Reply:
```ruby
return unless (verbose? || debug?) && !quiet?
```

Another option to match the unless in `debug`. Reads a little nicer maybe.

Reply:
both of those mean we do not print anything by default on `warn`, which I did not want to change - there would be test failures that rely on `warn`ings being printed with the default configuration (`verbose?`, `debug?`, `quite?` all return `nil`) thus the slightly more complicated condition...

Reply:
Aha ok... yeah let's preserve the current behavior exactly.